### DISCUSSION TITLE
Unknown (FileSystemError): Error: EROFS: read-only file system

### DISCUSSION THREAD

**Discussion by pikooli** [microsoft/vscode-discussions#401] [Extension Development QnA] [answered]
Hello, I try to create a Vscode extension but I have some difficulty to make test for some function of vscode.

My code look like this :

```
import { messages } from "../I18n";
import { Uri, workspace, WorkspaceEdit } from "vscode";
import { checkIsAFile, getCurrentWorkspacePath } from "./index";
import * as path from "path";

export const createFile = async (filePath: string) => {
  const wsPath = getCurrentWorkspacePath();
  const wsedit = new WorkspaceEdit();

  if (!filePath) {
    throw messages.errors.creatingFile;
  }
  const completeFilePath = path.join(wsPath, filePath);
  if (await checkIsAFile(completeFilePath)) {
    throw messages.errors.fileAlreadyExist;
  }
  const fileUri = Uri.file(completeFilePath);
  wsedit.createFile(fileUri);
  await workspace.applyEdit(wsedit);
  return completeFilePath;
};
```
 
The  ` wsedit.createFile(fileUri);` and ` await workspace.applyEdit(wsedit);` don't execute inside of mocha, 

It give me this error : 
```
Unable to write file '/test/path/file.txt' (Unknown (FileSystemError): Error: EROFS: read-only file system, mkdir '/test')
IGNORING workspace edit: Error: Unable to write file '/test/path/file.txt' (Unknown (FileSystemError): Error: EROFS: read-only file system, mkdir '/test')
```

Does someone have a idea ? 

Repo of my code : 
https://github.com/pikooli/create-file-snippet

**Comment by r3m0t** [comment, marked as answer]
Run `mount` to check, maybe the filesystem on the host really is readonly. You can use `os.tmpdir()` to get a path to a writable folder. Or in the extension, `ExtensionContext.logUri`.

**Comment by maxsyst** [comment]
<img width="2187" height="1052" alt="image" src="https://github.com/user-attachments/assets/63a65d8f-f38e-44ac-ab88-abae8d7bb700" />
I meet the same problem, excuse me , how to slove this problem? or change disk ???
